aten/src/ATen/native/Onehot.cpp

```diff
@@ -23,14 +23,14 @@ Tensor one_hot(const Tensor &self, int64_t num_classes) {
     }
 
     // non-empty tensor
-    if (self.device().type() != at::kCUDA) {
+    if (self.device().type() != at::kCUDA && self.device().type() != at::kMPS) {
```

Comment:
does MPS implementation throw an error if index is out of range?

Reply:
That's a good point, we don't raise an error. But, currently these `item()` reads are causing syncs on GPU which is leading to perf issues. Do we have an option to avoid these ?

Reply:
@ngimel , these checks are expensive because of the syncs. We can add checks where possible in the corresponding ops on CPU. But we don't raise an error after we issue the dispatch.